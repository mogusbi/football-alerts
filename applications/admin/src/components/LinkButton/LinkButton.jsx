import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

function LinkButton ({to, ...props}) {
  return (
    <Button
      component={Link}
      to={to}
      {...props}
    />
  );
}

LinkButton.propTypes = {
  to: PropTypes.string.isRequired
};

export default memo(LinkButton);
