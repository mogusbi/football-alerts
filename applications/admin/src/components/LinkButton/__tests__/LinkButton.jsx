import {mount} from 'enzyme';
import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import LinkButton from '../LinkButton';

describe('LinkButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <MemoryRouter>
        <LinkButton to='/somewhere'>
          A link button
        </LinkButton>
      </MemoryRouter>
    );
  });

  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
