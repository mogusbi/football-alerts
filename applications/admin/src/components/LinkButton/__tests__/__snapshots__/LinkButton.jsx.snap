// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton should render component 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {
          "pathname": "",
        },
      }
    }
  >
    <LinkButton
      to="/somewhere"
    >
      <WithStyles(Button)
        component={[Function]}
        to="/somewhere"
      >
        <Button
          classes={
            Object {
              "colorInherit": "MuiButton-colorInherit-22",
              "contained": "MuiButton-contained-12",
              "containedPrimary": "MuiButton-containedPrimary-13",
              "containedSecondary": "MuiButton-containedSecondary-14",
              "disabled": "MuiButton-disabled-21",
              "extendedFab": "MuiButton-extendedFab-19",
              "fab": "MuiButton-fab-18",
              "flat": "MuiButton-flat-6",
              "flatPrimary": "MuiButton-flatPrimary-7",
              "flatSecondary": "MuiButton-flatSecondary-8",
              "focusVisible": "MuiButton-focusVisible-20",
              "fullWidth": "MuiButton-fullWidth-26",
              "label": "MuiButton-label-2",
              "mini": "MuiButton-mini-23",
              "outlined": "MuiButton-outlined-9",
              "outlinedPrimary": "MuiButton-outlinedPrimary-10",
              "outlinedSecondary": "MuiButton-outlinedSecondary-11",
              "raised": "MuiButton-raised-15",
              "raisedPrimary": "MuiButton-raisedPrimary-16",
              "raisedSecondary": "MuiButton-raisedSecondary-17",
              "root": "MuiButton-root-1",
              "sizeLarge": "MuiButton-sizeLarge-25",
              "sizeSmall": "MuiButton-sizeSmall-24",
              "text": "MuiButton-text-3",
              "textPrimary": "MuiButton-textPrimary-4",
              "textSecondary": "MuiButton-textSecondary-5",
            }
          }
          color="default"
          component={[Function]}
          disableFocusRipple={false}
          disabled={false}
          fullWidth={false}
          mini={false}
          size="medium"
          to="/somewhere"
          type="button"
          variant="text"
        >
          <WithStyles(ButtonBase)
            className="MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
            component={[Function]}
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="MuiButton-focusVisible-20"
            to="/somewhere"
            type="button"
          >
            <ButtonBase
              centerRipple={false}
              className="MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
              classes={
                Object {
                  "disabled": "MuiButtonBase-disabled-28",
                  "focusVisible": "MuiButtonBase-focusVisible-29",
                  "root": "MuiButtonBase-root-27",
                }
              }
              component={[Function]}
              disableRipple={false}
              disableTouchRipple={false}
              disabled={false}
              focusRipple={true}
              focusVisibleClassName="MuiButton-focusVisible-20"
              tabIndex="0"
              to="/somewhere"
              type="button"
            >
              <Link
                className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
                to="/somewhere"
              >
                <a
                  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
                  href=""
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <span
                    className="MuiButton-label-2"
                  >
                    A link button
                  </span>
                  <NoSsr
                    defer={false}
                    fallback={null}
                  >
                    <WithStyles(TouchRipple)
                      center={false}
                      innerRef={[Function]}
                    >
                      <TouchRipple
                        center={false}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child-34",
                            "childLeaving": "MuiTouchRipple-childLeaving-35",
                            "childPulsate": "MuiTouchRipple-childPulsate-36",
                            "ripple": "MuiTouchRipple-ripple-31",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate-33",
                            "rippleVisible": "MuiTouchRipple-rippleVisible-32",
                            "root": "MuiTouchRipple-root-30",
                          }
                        }
                      >
                        <TransitionGroup
                          childFactory={[Function]}
                          className="MuiTouchRipple-root-30"
                          component="span"
                          enter={true}
                          exit={true}
                        >
                          <span
                            className="MuiTouchRipple-root-30"
                          />
                        </TransitionGroup>
                      </TouchRipple>
                    </WithStyles(TouchRipple)>
                  </NoSsr>
                </a>
              </Link>
            </ButtonBase>
          </WithStyles(ButtonBase)>
        </Button>
      </WithStyles(Button)>
    </LinkButton>
  </Router>
</MemoryRouter>
`;
